From 03d1e6089230bad79b78ce6e8ea2b872cbaf37e2 Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@redhat.com>
Date: Mon, 29 Feb 2016 09:48:12 +0100
Subject: [PATCH 1/8] opencdk: Fixes to prevent undefined behavior (found with
 libubsan)

---
 lib/opencdk/misc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/opencdk/misc.c b/lib/opencdk/misc.c
index 0d4ee8912..35172e5dd 100644
--- a/lib/opencdk/misc.c
+++ b/lib/opencdk/misc.c
@@ -41,7 +41,7 @@ u32 _cdk_buftou32(const byte * buf)
 
 	if (!buf)
 		return 0;
-	u = buf[0] << 24;
+	u = ((u32)buf[0]) << 24;
 	u |= buf[1] << 16;
 	u |= buf[2] << 8;
 	u |= buf[3];
-- 
2.11.0

