From 5888e3cc63611396adb90d3ad1dc42a0bdb5850b Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@redhat.com>
Date: Mon, 20 Feb 2017 11:14:49 +0100
Subject: [PATCH] opencdk/read-packet.c: corrected typo in type cast

Signed-off-by: Nikos Mavrogiannopoulos <nmav@redhat.com>
---
 lib/opencdk/read-packet.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/opencdk/read-packet.c b/lib/opencdk/read-packet.c
index d95845d56..a2631fed1 100644
--- a/lib/opencdk/read-packet.c
+++ b/lib/opencdk/read-packet.c
@@ -919,7 +919,7 @@ read_new_length(cdk_stream_t inp,
 		*r_len = ((c - 192) << 8) + c1 + 192;
 	} else if (c == 255) {
 		c1 = read_32(inp);
-		if (c1 == (u32)-1) {
+		if (c1 == -1) {
 			return;
 		}
 
-- 
2.11.0

